app/controllers/smart_answers_controller.rb

```diff
@@ -53,7 +53,8 @@ def visualise
   def find_smart_answer
     @name = params[:id].to_sym
     @smart_answer = flow_registry.find(@name.to_s)
-    @presenter = FlowPresenter.new(request.params, @smart_answer)
+    current_state = SmartAnswer::ResolveState.new(@smart_answer).from_params(request.params)
```

Comment:
Yeah that's a good shout. I've changed that and renamed current_state in FlowPresenter to state since there aren't any other states that needs to consider either.